lib/ClangImporter/ClangImporter.cpp

```diff
@@ -3242,6 +3243,32 @@ void ClangImporter::verifyAllModules() {
 #endif
 }
 
+const clang::Type *
+ClangImporter::parseClangFunctionType(StringRef typeStr) const {
+  auto &sema = Impl.getClangSema();
+  // TODO: Pass down the filename and SourceLoc for this call.
+  auto parsedType = sema.ParseTypeFromStringCallback(typeStr,
+                                                     "<cType parsing>", {});
+  if (!parsedType.isUsable())
+    return nullptr;
+  clang::QualType resultType = clang::Sema::GetTypeFromParser(parsedType.get());
+  auto *typePtr = resultType.getTypePtrOrNull();
+  if (typePtr && typePtr->getAs<clang::FunctionType>())
+    return typePtr;
+  return nullptr;
+}
+
+void ClangImporter::printClangType(const clang::Type *type,
+                                   llvm::raw_ostream &os) const {
+  auto policy = clang::PrintingPolicy(getClangASTContext().getLangOpts());
+  SmallString<64> buf;
+  {
+    llvm::raw_svector_ostream os_tmp(buf);
+    clang::QualType(type, 0).print(os_tmp, policy);
+  }
+  os << QuotedString(StringRef(buf));
```

Comment:
I think this method probably ought to just print into the stream, and the caller can add quotes if it wants.